unmute/main_websocket.py

```diff
@@ -80,8 +81,43 @@
     Annotated[ora.ClientEvent, Field(discriminator="type")]
 )
 
-# Allow CORS for local development
-CORS_ALLOW_ORIGINS = ["http://localhost", "http://localhost:3000"]
+DEFAULT_CORS_ALLOW_ORIGINS = ["http://localhost", "http://localhost:3000"]
+
+
+def _normalize_origin(origin: str | None) -> str | None:
+    if origin is None:
+        return None
+    return origin.rstrip("/")
+
+
+def _load_cors_allow_origins() -> list[str]:
+    raw_env = os.getenv("UNMUTE_CORS_ALLOW_ORIGINS")
+    defaults = [_normalize_origin(origin) for origin in DEFAULT_CORS_ALLOW_ORIGINS]
+
+    if not raw_env:
+        return [origin for origin in defaults if origin]
+
+    env_values = [
+        _normalize_origin(value.strip())
+        for value in raw_env.split(",")
+        if value.strip()
+    ]
+
+    # Preserve order while removing duplicates and skipping None.
+    seen = set()
+    merged: list[str] = []
+    for origin in [*defaults, *env_values]:
+        if not origin or origin in seen:
+            continue
+        seen.add(origin)
+        merged.append(origin)
+
+    return merged
+
+
+CORS_ALLOW_ORIGINS = _load_cors_allow_origins()
+CORS_ALLOW_ORIGINS_SET = set(CORS_ALLOW_ORIGINS)
+logger.info("Configured CORS allow origins: %s", CORS_ALLOW_ORIGINS)
```

Comment:
Without this, the front-end on port 3000 cannot connect to the backend on port 8000, because of the way runpod assigns custom domains for the port forwarding.